client/src/main/com/sinch/sdk/auth/adapters/ApplicationAuthManager.java

```diff
@@ -97,4 +99,49 @@ private String encode(String stringToSign) {
       throw new ApiAuthException(e);
     }
   }
+
+  public boolean validateAuthenticatedRequest(
+      String method, String path, Map<String, String> headers, String jsonPayload) {
+
+    // convert header keys to use case-insensitive map keys
+    Map<String, String> caseInsensitiveHeaders = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+    caseInsensitiveHeaders.putAll(headers);
+
+    String authorizationHeader = caseInsensitiveHeaders.get("Authorization");
+
+    // no authorization required
+    if (null == authorizationHeader) {
+      return true;
+    }
+
+    String[] split = authorizationHeader.split(" ");
+    String authorizationHash = split.length > 1 ? split[1] : "";
+
+    String computedHash = computeHash(caseInsensitiveHeaders, method, path, jsonPayload);
+
+    return computedHash.equals(authorizationHash);
```

Comment:
You are checking the hash, but the first thing to check would be that the application key (first part of the header) matches

Reply:
🤦